File: 02Fabryka/Students/wdenejko/Fabryka/CarFactory.cs
using System;
namespace Fabryka
{
	public class CarFactory
	{
		public Car CreateCar(String type)
		{
			Car samochod = null;

			if (type.Equals("audi"))
			{
				samochod = new Audi();
			}
			else if (type.Equals("vw"))
			{
				samochod = new Vw();
			}

			return samochod;
		}
	}
}

File: 02Fabryka/Students/dzduniak/fabryka/CarFactory.java
package fabryka;

import java.util.HashMap;
import java.util.Map;
import java.util.function.Supplier;

// Dopisz opis i wyślij
public class CarFactory {
    private final Map<String, Supplier<Car>> suppliers;

    public CarFactory() {
        suppliers = new HashMap<>();
    }

    public void register(String id, Supplier<Car> supplier) {
        suppliers.put(id, supplier);
    }

    public Car createCar(String id) {
        Supplier<Car> s = suppliers.get(id);
        if (s != null)
            return s.get();

        return null;
    }
}
